### DISCUSSION TITLE
Azure Functions Core Tools Installation on Raspberry Pi running Ubuntu Server

### DISCUSSION THREAD

**Discussion by mr-chris-reed** [microsoft/IoT-For-Beginners#431] [Q&A]
Hi there,

I am working through Lesson 9 and have run into a bit of a roadblock in that I am having difficulty installing the Azure Functions Core Tools.  I posted my question on Stack Overflow with as many details as I could.

https://stackoverflow.com/questions/74644088/installing-azure-functions-core-tools-on-raspberry-pi-running-ubuntu-server

But, I thought I would also try to alert anyone who may be monitoring the discussions here.

Any insight would be greatly appreciated.

Thank you.

**Comment by mr-chris-reed** [comment]
@jimbobbennett - any thoughts regarding Azure Functions Core Tools running on Raspberry Pi hardware?

Thanks:)

**Comment by tyeth** [comment]
Could try the npm version https://www.npmjs.com/package/azure-functions-core-tools

**Reply by mr-chris-reed** [reply]
Hi @tyeth - thanks for the suggestion.  I did try to use npm to install the Azure Functions Core Tools.  I don't remember the exact error I received, but couldn't get it to work.

**Comment by jimbobbennett** [comment]
Unfortunately there is no linux build for Arm64. There is an [issue raised for this](https://github.com/Azure/azure-functions-core-tools/issues/3112), but nothing has been done on it.

**Reply by mr-chris-reed** [reply]
Okay, thank you for the info.  I think we may have gotten around our issue by spinning up some small VMs in Azure and running Azure CLI and Azure Functions Core Tools from there.  It would be nice if we could eventually do everything on our Pi, as it would keep our toolset small.  We are currently using school-issued Chromebooks, Raspberry Pi 4B, and the Seeed starter kit.  The course has been really great so far.  Thanks @jimbobbennett!
